src/com/google/enterprise/adaptor/fs/WindowsFileDelegate.java

```diff
@@ -425,7 +449,8 @@ public MonitorThread(Path watchPath, AsyncDocIdPusher pusher,
       this.watchPath = watchPath;
       this.pusher = pusher;
       this.startSignal = startSignal;
-      stopEvent = Kernel32.INSTANCE.CreateEvent(null, false, false, null);
+      stopEvent = kernel32.CreateEvent(null, false, false, null);
```

Comment:
Is this the part where the handle needs to be shared more broadly so you can wait on it outside the monitor loop?